Frontend/ui-driver/src/Screens/ReferralScreen/Controller.purs

```diff
@@ -143,13 +145,15 @@ data ScreenOutput = GoToHomeScreen
 eval :: Action -> ReferralScreenState -> Eval Action ScreenOutput ReferralScreenState
 
 eval (UpdateLeaderBoard (LeaderBoardRes leaderBoardRes)) state = do
+  _ <- pure $ firebaseLogEvent "ny_driver_leaderboard"
   let dataLength = DA.length leaderBoardRes.driverList
       rankersData = sortWith (_.rank) (transformLeaderBoardList (filter (\(DriversInfo info) -> info.rank <= 10 && info.totalRides > 0 && info.rank > 0) leaderBoardRes.driverList)) <> (replicate (10 - dataLength) RSD.dummyRankData)
       currentDriverData = case (filter (\(DriversInfo info) -> info.isCurrentDriver && info.rank > 0 && info.totalRides > 0) leaderBoardRes.driverList) !! 0 of
                             Just driverData -> transformLeaderBoard driverData
                             Nothing         -> RSD.dummyRankData
       lastUpdatedAt = convertUTCtoISC (fromMaybe (getCurrentUTC "") leaderBoardRes.lastUpdatedAt) "h:mm A"
   let newState = state{ props { rankersData = rankersData, currentDriverData = currentDriverData, showShimmer = false, noData = not (dataLength > 0), lastUpdatedAt = lastUpdatedAt } }
+  _ <- pure $ setRefreshing "2000034" false
```

Comment:
@Anuragini-2506 Can you get if from `getNewIDWithTag` function , instead of hardcoding

Reply:
used getNewIDWithTag function for getting the id.